### ISSUE TITLE
Build failure: virtualbox (after libxml2 update)

### ISSUE THREAD

**Issue by spiage** [NixOS/nixpkgs#280591]
### Steps To Reproduce

1. build *virtualbox* from master branch with:
virtualisation.virtualbox.host.enable = true;

### Build log

```
kmk: *** Exiting with status 2
kBuild: Compiling RuntimeR3 - /build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp: In constructor 'xml::XmlParserBase::XmlParserBase()':
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:1873:37: error: invalid conversion from 'void (*)(void*, xmlErrorPtr) noexcept' {aka 'void (*)(void*, _xmlError*)'} to 'xmlStructuredErrorFunc' {aka 'void (*>
 1873 |     xmlSetStructuredErrorFunc(NULL, xmlParserBaseStructuredError);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                     |
      |                                     void (*)(void*, xmlErrorPtr) noexcept {aka void (*)(void*, _xmlError*)}
In file included from /nix/store/iqjsxkcdnvvz1bfpq960ygicc5clz9hv-libxml2-2.12.3-unstable-2023-12-14-dev/include/libxml2/libxml/valid.h:15,
                 from /nix/store/iqjsxkcdnvvz1bfpq960ygicc5clz9hv-libxml2-2.12.3-unstable-2023-12-14-dev/include/libxml2/libxml/parser.h:19,
                 from /nix/store/iqjsxkcdnvvz1bfpq960ygicc5clz9hv-libxml2-2.12.3-unstable-2023-12-14-dev/include/libxml2/libxml/tree.h:17,
                 from /build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:54:
/nix/store/iqjsxkcdnvvz1bfpq960ygicc5clz9hv-libxml2-2.12.3-unstable-2023-12-14-dev/include/libxml2/libxml/xmlerror.h:898:57: note:   initializing argument 2 of 'void xmlSetStructuredErrorFunc(void*, xmlStructur>
  898 |                                  xmlStructuredErrorFunc handler);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp: In member function 'void xml::XmlMemParser::read(const void*, size_t, const RTCString&, xml::Document&)':
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:1934:43: error: invalid conversion from 'const xmlError*' {aka 'const _xmlError*'} to 'xmlErrorPtr' {aka '_xmlError*'} [-fpermissive]
 1934 |         throw XmlError(xmlCtxtGetLastError(m_ctxt));
      |                        ~~~~~~~~~~~~~~~~~~~^~~~~~~~
      |                                           |
      |                                           const xmlError* {aka const _xmlError*}
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:134:32: note:   initializing argument 1 of 'xml::XmlError::XmlError(xmlErrorPtr)'
  134 | XmlError::XmlError(xmlErrorPtr aErr)
      |                    ~~~~~~~~~~~~^~~~
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp: In member function 'void xml::XmlFileParser::read(const RTCString&, xml::Document&)':
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:2194:43: error: invalid conversion from 'const xmlError*' {aka 'const _xmlError*'} to 'xmlErrorPtr' {aka '_xmlError*'} [-fpermissive]
 2194 |         throw XmlError(xmlCtxtGetLastError(m_ctxt));
      |                        ~~~~~~~~~~~~~~~~~~~^~~~~~~~
      |                                           |
      |                                           const xmlError* {aka const _xmlError*}
/build/VirtualBox-7.0.12/src/VBox/Runtime/r3/xml.cpp:134:32: note:   initializing argument 1 of 'xml::XmlError::XmlError(xmlErrorPtr)'
  134 | XmlError::XmlError(xmlErrorPtr aErr)
      |                    ~~~~~~~~~~~~^~~~
kmk: *** [/build/VirtualBox-7.0.12/kBuild/footer-pass2-compiling-targets.kmk:277: /build/VirtualBox-7.0.12/out/linux.amd64/release/obj/RuntimeR3/r3/xml.o] Error 1
```

### Notify maintainers

@sandro (or @SuperSandro2000 ?)

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
spiage@nixos:~$ nix-shell -p nix-info --run "nix-info -m"
these 6 paths will be fetched (0.75 MiB download, 10.19 MiB unpacked):
  /nix/store/hnz8p5w9xc0ivsiyrgky6f6h7sjb28a5-file-5.45
  /nix/store/y83pz60av31hpr11zlbvc7bb93bk5fkd-gnu-config-2023-09-19
  /nix/store/452v8bl3sjaq4kil1d9vqxc5vns4cvpp-gnumake-4.4.1
  /nix/store/pyq6gyhgck1nkfyjs6842ysxkxzjxkaj-patchelf-0.15.0
  /nix/store/d4jf1cbbk494zwgbqz31pxgigpsbh6w2-stdenv-linux
  /nix/store/j0blrlhddcnqgq1v1cd4pfggjr2wah8n-update-autotools-gnu-config-scripts-hook
copying path '/nix/store/y83pz60av31hpr11zlbvc7bb93bk5fkd-gnu-config-2023-09-19' from 'https://cache.nixos.org'...
copying path '/nix/store/hnz8p5w9xc0ivsiyrgky6f6h7sjb28a5-file-5.45' from 'https://cache.nixos.org'...
copying path '/nix/store/452v8bl3sjaq4kil1d9vqxc5vns4cvpp-gnumake-4.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/pyq6gyhgck1nkfyjs6842ysxkxzjxkaj-patchelf-0.15.0' from 'https://cache.nixos.org'...
copying path '/nix/store/j0blrlhddcnqgq1v1cd4pfggjr2wah8n-update-autotools-gnu-config-scripts-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/d4jf1cbbk494zwgbqz31pxgigpsbh6w2-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0, NixOS, 24.05 (Uakari), 24.05.20240112.0dc9c89`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - nixpkgs: `/nix/store/bsmx16hp1rh0pfvs76k3jf109nf3r3l5-source`
 ```

**Comment by SuperSandro2000** [comment]
Either we update virtualbox and hope this got fixed, we find a patch or pin an older libxml2 version.

**Comment by wegank** [comment]
http://git.pld-linux.org/?p=packages/VirtualBox.git;a=blob;f=libxml2.12.patch;h=60df8c8b04df1b982badca13d867a925e2628fd5;hb=a4c810d

**Comment by spiage** [comment]
Solved by #280675
